### PR TITLE
fix 2 failing unit tests across IE9 - IE11

### PR DESCRIPTION
**Pull request by bitfrost**

### CODE REVIEW

**Comment by kof** [comment]
wow interesting, I will test it soon, thanks!

File: package.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "karma-sourcemap-loader": "0.3.5",
     "karma-webpack": "1.7.0",
     "lodash": "3.10.1",
+    "qunitjs": "^1.20.0",
```
Comment (kof) [inline package.json]: qunitjs was removed from deps because I started to use karma .... why did you add it back?

Comment (bitfrost) [inline package.json]: https://github.com/jsstyles/jss/blob/master/karma.conf.js#L11
You still have a dependency on it, without it installed:   (I am using npm3, so I don't get peerDependencies installed automatically)
Error: Cannot find module 'qunitjs'
    at Function.Module._resolveFilename (module.js:337:15)
    at Function.require.resolve (module.js:389:19)
    at initQUnit (/Users/kobrien/dev/c/jss/node_modules/karma-qunit/lib/index.j

Comment (kof) [inline package.json]: oh I didn't try that with npm3, you are possibly right.

**Comment by kof** [comment]
Which 2 errors did you get?

15.Rule: applyTo (1, 1, 2)
and
16.Rule: applyTo with array value (1, 0, 1)

**Comment by kof** [comment]
Please try again master state. Your issue should have been fixed.
Test suite runs now on IE8+. If something fails please add a test, you can run it also using test/index.html now. You only need `npm i && npm build`

**Comment by bitfrost** [comment]
Thank you for the follow up,

The pull request was opened because I encountered failing unit tests when running npm test via your browerstack config.  If I see other issues not covered by tests I will add them.

I just  cloned master with your recent changes and I did a fresh npm install.  I have my own browerstack account setup via environment vars.   Here is a log from my console I am getting more failures than before including some elint issues during npm install.

terminal log after clone:

hqkobrien902:jss kobrien$ npm install
npm WARN engine karma@0.13.13: wanted: {"node":"0.10 || 0.12 || 4"} (current: {"node":"5.0.0","npm":"3.3.9"})
npm WARN deprecated babel-core@5.8.25: Babel 5 is no longer being maintained. Upgrade to Babel 6.

> fsevents@1.0.5 install /Users/kobrien/dev/jss/node_modules/fsevents
> node-pre-gyp install --fallback-to-build

[fsevents] Success: "/Users/kobrien/dev/jss/node_modules/fsevents/lib/binding/Release/node-v47-darwin-x64/fse.node" is installed via remote

> bufferutil@1.2.1 install /Users/kobrien/dev/jss/node_modules/bufferutil
> node-gyp rebuild

  CXX(target) Release/obj.target/bufferutil/src/bufferutil.o
  SOLINK_MODULE(target) Release/bufferutil.node

> utf-8-validate@1.2.1 install /Users/kobrien/dev/jss/node_modules/utf-8-validate
> node-gyp rebuild

  CXX(target) Release/obj.target/validation/src/validation.o
  SOLINK_MODULE(target) Release/validation.node

> jss@3.0.0 prepublish /Users/kobrien/dev/jss
> npm run all && git push --tags


> jss@3.0.0 all /Users/kobrien/dev/jss
> npm run lint && npm run test && npm run build


> jss@3.0.0 lint /Users/kobrien/dev/jss
> eslint ./src ./test


/Users/kobrien/dev/jss/src/DomRenderer.js
  12:7   error  Closing curly brace appears on the same line as the subsequent block  brace-style
  12:19  error  Empty block statement                                                 no-empty
  16:18  error  Multiple spaces found before 'document'                               no-multi-spaces

/Users/kobrien/dev/jss/test/Rule.js
  2:13  error  "utils" is defined but never used  no-unused-vars

✖ 4 problems (4 errors, 0 warnings)


npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/Cellar/node/5.0.0/bin/node" "/usr/local/bin/npm" "run" "lint"
npm ERR! node v5.0.0
npm ERR! npm  v3.3.9
npm ERR! code ELIFECYCLE
npm ERR! jss@3.0.0 lint: `eslint ./src ./test`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jss@3.0.0 lint script 'eslint ./src ./test'.
npm ERR! This is most likely a problem with the jss package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint ./src ./test
npm ERR! You can get their info via:
npm ERR!     npm owner ls jss
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/kobrien/dev/jss/npm-debug.log

npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/Cellar/node/5.0.0/bin/node" "/usr/local/bin/npm" "run" "all"
npm ERR! node v5.0.0
npm ERR! npm  v3.3.9
npm ERR! code ELIFECYCLE
npm ERR! jss@3.0.0 all: `npm run lint && npm run test && npm run build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jss@3.0.0 all script 'npm run lint && npm run test && npm run build'.
npm ERR! This is most likely a problem with the jss package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run lint && npm run test && npm run build
npm ERR! You can get their info via:
npm ERR!     npm owner ls jss
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/kobrien/dev/jss/npm-debug.log

npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/Cellar/node/5.0.0/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v5.0.0
npm ERR! npm  v3.3.9
npm ERR! code ELIFECYCLE
npm ERR! jss@3.0.0 prepublish: `npm run all && git push --tags`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the jss@3.0.0 prepublish script 'npm run all && git push --tags'.
npm ERR! This is most likely a problem with the jss package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run all && git push --tags
npm ERR! You can get their info via:
npm ERR!     npm owner ls jss
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/kobrien/dev/jss/npm-debug.log
hqkobrien902:jss kobrien$ git branch
* master
hqkobrien902:jss kobrien$ export USE_CLOUD=1
hqkobrien902:jss kobrien$ npm test

> jss@3.0.0 test /Users/kobrien/dev/jss
> karma start

11 11 2015 22:51:35.344:INFO [karma]: Karma v0.13.13 server started at http://localhost:9876/
BrowserStackTunnel: binary not present
11 11 2015 22:51:35.368:INFO [launcher]: Starting browser safari 8.0 (OS X Yosemite) on BrowserStack
11 11 2015 22:51:35.369:ERROR [launcher]: [TypeError: browser.on is not a function]
TypeError: browser.on is not a function
    at /Users/kobrien/dev/jss/node_modules/karma/lib/launcher.js:94:17
    at next (/Users/kobrien/dev/jss/node_modules/batch/index.js:118:7)
    at Batch.end (/Users/kobrien/dev/jss/node_modules/batch/index.js:154:5)
    at launch (/Users/kobrien/dev/jss/node_modules/karma/lib/launcher.js:110:11)
    at invoke (/Users/kobrien/dev/jss/node_modules/di/lib/injector.js:75:15)
    at Server.<anonymous> (/Users/kobrien/dev/jss/node_modules/karma/lib/server.js:170:24)
    at Server.g (events.js:260:16)
    at emitNone (events.js:67:13)
    at Server.emit (events.js:166:7)
    at emitListeningNT (net.js:1260:10)
    at doNTCallback1 (node.js:439:9)
    at process._tickCallback (node.js:361:17)
11 11 2015 22:51:35.371:INFO [launcher]: Starting browser iphone (ios 7.0) on BrowserStack
11 11 2015 22:51:35.371:INFO [launcher]: Starting browser chrome 39.0 (Windows 8.1) on BrowserStack
11 11 2015 22:51:35.371:INFO [launcher]: Starting browser firefox 32.0 (Windows 8.1) on BrowserStack
11 11 2015 22:51:35.371:INFO [launcher]: Starting browser ie 8.0 (Windows 8.1) on BrowserStack
11 11 2015 22:51:35.371:INFO [launcher]: Starting browser ie 9.0 (Windows 8.1) on BrowserStack
11 11 2015 22:51:35.372:INFO [launcher]: Starting browser ie 10.0 (Windows 8) on BrowserStack
11 11 2015 22:51:35.372:INFO [launcher]: Starting browser ie 11.0 (Windows 8.1) on BrowserStack
BrowserStackTunnel: download binary for darwin-x64 ...
BrowserStackTunnel: download complete
BrowserStackTunnel: chmod 0755 binary
11 11 2015 22:51:42.586:ERROR [launcher.browserstack]: Can not start ie 8.0 (Windows 8.1)
  Error: Validation Failed - `browser_version` invalid
11 11 2015 22:51:42.682:ERROR [launcher.browserstack]: Can not start ie 9.0 (Windows 8.1)
  Error: Validation Failed - `browser_version` invalid
11 11 2015 22:51:50.311:INFO [Safari 8.0.8 (Mac OS X 10.10.5)]: Connected on socket d-ivDwhcSu8ekS27AAAA with id 59329418
Safari 8.0.8 (Mac OS X 10.10.5) StyleSheet link FAILED
	failed, expected argument to be truthy, was: false
	Expected: true
	Actual: false
	/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:2145:18
	/Users/kobrien/dev/jss/tests.webpack.js:1644:6 <- webpack:///test/StyleSheet.js:137:5
	runTest@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:32
	run@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:11
	/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:14
	process@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:24
	begin@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:673:9
	/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:733:9

Safari 8.0.8 (Mac OS X 10.10.5): Executed 39 of 39 (1 FAILED) (0.907 secs / 0.013 secs)
11 11 2015 22:51:52.733:INFO [launcher.browserstack]: safari 8.0 (OS X Yosemite) session at https://www.browserstack.com/automate/builds/1acfa6125ec199ffd784517c8bc946002896bf45/sessions/a66a8115eeb99a4fd1bdc622c90614212c284969
11 11 2015 22:51:52.844:INFO [launcher.browserstack]: ie 11.0 (Windows 8.1) session at https://www.browserstack.com/automate/builds/1acfa6125ec199ffd784517c8bc946002896bf45/sessions/92092b117b2424c450cab87e6747f0d74b929ddc
11 11 2015 22:51:52.847:INFO [launcher.browserstack]: chrome 39.0 (Windows 8.1) session at https://www.browserstack.com/automate/builds/1acfa6125ec199ffd784517c8bc946002896bf45/sessions/86614b6452209fbe55b32fc9ba33d94846de2da6
11 11 2015 22:51:52.848:INFO [launcher.browserstack]: firefox 32.0 (Windows 8.1) session at https://www.browserstack.com/automate/builds/1acfa6125ec199ffd784517c8bc946002896bf45/sessions/eda0458483322aafc91f607cc1b411220a9f6c43
11 11 2015 22:51:53.120:INFO [launcher.browserstack]: ie 10.0 (Windows 8) session at https://www.browserstack.com/automate/builds/1acfa6125ec199ffd784517c8bc946002896bf45/sessions/600099b62f91d8229b6e3ef3a3398e7594eeab3b
11 11 2015 22:51:53.589:INFO [launcher.browserstack]: iphone (ios 7.0) session at https://www.browserstack.com/automate/builds/1acfa6125ec199ffd784517c8bc946002896bf45/sessions/c88709ff3afb490a92b81ebd42c74a536143e0b5
Safari 8.0.8 (Mac OS X 10.10.5): Executed 39 of 39 (1 FAILED) (0.907 secs / 0.013 secs)
Firefox 32.0.0 (Windows 8.1 0.0.0) StyleSheet addRule/getRule on attached sheet FAILED
	Died on test #2 @/Users/kobrien/dev/jss/tests.webpack.js:1577:2 <- webpack:///test/StyleSheet.js:70:0
	__webpack_require__@/Users/kobrien/dev/jss/tests.webpack.js:30:12 <- webpack:/webpack/bootstrap c4ecdfa3710d1061e419:19:0
	webpackContext@/Users/kobrien/dev/jss/tests.webpack.js:73:3 <- webpack:/test \.js$:8:0
	@/Users/kobrien/dev/jss/tests.webpack.js:60:2 <- webpack:///tests.webpack.js:2:0
	__webpack_require__@/Users/kobrien/dev/jss/tests.webpack.js:30:12 <- webpack:/webpack/bootstrap c4ecdfa3710d1061e419:19:0
	@/Users/kobrien/dev/jss/tests.webpack.js:50:11 <- webpack:/webpack/bootstrap c4ecdfa3710d1061e419:39:0
	@/Users/kobrien/dev/jss/tests.webpack.js:1656:7 <- null:0:0
	webpackUniversalModuleDefinition@/Users/kobrien/dev/jss/tests.webpack.js:9:3 <- webpack:///webpack/universalModuleDefinition:9:0
	@/Users/kobrien/dev/jss/tests.webpack.js:10:1 <- webpack:///webpack/universalModuleDefinition:10:1
	: sheet.renderer.element.sheet.rules is undefined
	@/Users/kobrien/dev/jss/tests.webpack.js:1584:8 <- webpack:///test/StyleSheet.js:77:8
	runTest@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:4
	Test.prototype.run@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:4
	run/<@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:6
	process@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:4
	begin@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:673:2
	resumeProcessing/<@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:733:4


Firefox 32.0.0 (Windows 8.1 0.0.0) StyleSheet virtual rendering FAILED
	Expected: 0
	Actual: 1
	applyCurrent/<@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:2145:5
	@/Users/kobrien/dev/jss/tests.webpack.js:1651:4 <- webpack:///test/StyleSheet.js:144:2
	runTest@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:4
	Test.prototype.run@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:4
	run/<@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:6
	process@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:4
	begin@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:673:2
	resumeProcessing/<@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:733:4


Safari 8.0.8 (Mac OS X 10.10.5): Executed 39 of 39 (1 FAILED) (0.907 secs / 0.013 secs)
IE 10.0.0 (Windows 8 0.0.0) StyleSheet attach/detach FAILED
	Expected: .jss-0-17{float:left}
	Actual:
	   at Anonymous function (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:2145:5)
	   at Anonymous function (/Users/kobrien/dev/jss/tests.webpack.js:1571:4 <- webpack:///test/StyleSheet.js:64:2)
	   at runTest (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:4)
	   at run (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:4)
	   at Anonymous function (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:6)
	   at process (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:4)

IE 10.0.0 (Windows 8 0.0.0) StyleSheet toString unnamed FAILED
	Expected: a{width:1pxfloat:left}
	Actual:
	   at Anonymous function (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:2145:5)
	   at Anonymous function (/Users/kobrien/dev/jss/tests.webpack.js:1597:4 <- webpack:///test/StyleSheet.js:90:2)
	   at runTest (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:4)
	   at run (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:4)
	   at Anonymous function (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:6)
	   at process (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:4)

IE 10.0.0 (Windows 8 0.0.0) StyleSheet toString named FAILED
	Expected: .jss-0-0{width:1pxfloat:left}
	Actual:
	   at Anonymous function (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:2145:5)
	   at Anonymous function (/Users/kobrien/dev/jss/tests.webpack.js:1607:4 <- webpack:///test/StyleSheet.js:100:2)
	   at runTest (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:4)
	   at run (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:4)
	   at Anonymous function (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:6)
	   at process (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:4)

IE 10.0.0 (Windows 8 0.0.0) StyleSheet toString unnamed with media query FAILED
	Expected:
	a {
	  color: red;
	}
	@media (min-width: 1024px) {
	  a {
	    color: blue;
	  }
	}

	Actual:
	   at Anonymous function (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:2145:5)
	   at Anonymous function (/Users/kobrien/dev/jss/tests.webpack.js:1620:6 <- webpack:///test/StyleSheet.js:113:4)
	   at runTest (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:4)
	   at run (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:4)
	   at Anonymous function (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:6)
	   at process (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:4)

Safari 8.0.8 (Mac OS X 10.10.5): Executed 39 of 39 (1 FAILED) (0.907 secs / 0.013 secs)
Chrome 39.0.2171 (Windows 8.1 0.0.0) StyleSheet link FAILED
	failed, expected argument to be truthy, was: false
	Expected: true
	Actual: false
	    at /Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:2145:13
	    at Object.<anonymous> (/Users/kobrien/dev/jss/tests.webpack.js:1644:43 <- webpack:///test/StyleSheet.js:137:41)
	    at runTest (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:28)
	    at Object.Test.run (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:4)
	    at /Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:11
	    at process (/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:24)

Safari 8.0.8 (Mac OS X 10.10.5): Executed 39 of 39 (1 FAILED) (0.907 secs / 0.013 secs)
Mobile Safari 7.0.0 (iOS 7.0.0) StyleSheet link FAILED
	failed, expected argument to be truthy, was: false
	Expected: true
	Actual: false
	/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:2145:18
	/Users/kobrien/dev/jss/tests.webpack.js:1644:6 <- webpack:///test/StyleSheet.js:137:5
	runTest@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:903:32
	run@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:888:11
	/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:1030:14
	process@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:691:24
	begin@/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:673:9
	/Users/kobrien/dev/jss/node_modules/qunitjs/qunit/qunit.js:733:9

Safari 8.0.8 (Mac OS X 10.10.5): Executed 39 of 39 (1 FAILED) (0.907 secs / 0.013 secs)
IE 11.0.0 (Windows 8.1 0.0.0): Executed 39 of 39 SUCCESS (0.198 secs / 0.025 secs)
Firefox 32.0.0 (Windows 8.1 0.0.0): Executed 39 of 39 (2 FAILED) (0.194 secs / 0.02 secs)
IE 10.0.0 (Windows 8 0.0.0): Executed 39 of 39 (4 FAILED) (0.35 secs / 0.04 secs)
Chrome 39.0.2171 (Windows 8.1 0.0.0): Executed 39 of 39 (1 FAILED) (0.509 secs / 0.018 secs)
Mobile Safari 7.0.0 (iOS 7.0.0): Executed 39 of 39 (1 FAILED) (0.173 secs / 0.049 secs)
TOTAL: 9 FAILED, 225 SUCCESS
npm ERR! Test failed.  See above for more details.

**Comment by bitfrost** [comment]
Looks like you just fixed the eslint issue as I typed my previous comment :)

**Comment by kof** [comment]
try again)

**Comment by kof** [comment]
Also I am getting 
```
12 11 2015 05:25:21.542:ERROR [launcher]: [TypeError: browser.on is not a function]
TypeError: browser.on is not a function
```

**Comment by kof** [comment]
ok now it runs through all tests, thanks

**Comment by bitfrost** [comment]
Yep, looking good and IE8 support.  Sweet.
